aderyn_core/src/detect/experimental/immediate_parent.rs

```diff
@@ -45,6 +45,10 @@ impl IssueDetector for ImmediateParentDemonstrator {
                                 assert!(first_parent
                                     .appears_after(context, second_parent)
                                     .unwrap());
+                                assert!(first_parent
+                                    .appears_after(context, for_statement)
+                                    .unwrap());
+                                assert!(block.appears_after(context, for_statement).unwrap());
```

Comment:
So now `appears_after` and `appears_before` works on any general node and even AST Node (or combo)